src/conversation.ts

```diff
@@ -57,11 +57,12 @@ export type ConversationFlavor<C extends Context | undefined = undefined> =
     & { conversation: ConversationControls }
     & (
         C extends Context ? 
-                & C
-                // deno-lint-ignore no-explicit-any
-                & (C extends { session: PromiseLike<any> }
-                    ? LazySessionFlavor<ConversationSessionData>
-                    : SessionFlavor<ConversationSessionData>)
+                & Omit<C, "session">
+                & (C extends LazySessionFlavor<infer V>
+                    ? LazySessionFlavor<ConversationSessionData & V>
+                    : C extends SessionFlavor<infer V>
+                        ? SessionFlavor<ConversationSessionData & V>
```

Comment:
I don't think you need this operation with regular sessions

Reply:
How else would you install the correct session types again?